### ISSUE TITLE
Updating from 5.0.4 to 5.0.6 the dropdown page stopped beeing full screen when keyboard is open, downgrading for now

### ISSUE THREAD

**Issue by mariopepe** [salim-lachdhaf/searchable_dropdown#551]
This is how it was before (5.0.4) GOOD:

https://user-images.githubusercontent.com/14803862/223407956-ade4c5e4-6b3f-4df7-b118-73a6377c388f.mov

---

This is how it is now (5.0.6) BAD:

https://user-images.githubusercontent.com/14803862/223408213-8e7f68b5-e9ab-481d-ae41-df68f02a1c58.mov

**Comment by mariopepe** [comment]
actually 5.0.5 works well, there is a bug only in 5.0.6 that I do not even find in the github source code as I only find 5.0.5 in the changelog.md....

Weird

**Comment by salim-lachdhaf** [comment]
did you tried to use `useSafeArea` to `false` for you modal property ?

**Comment by mariopepe** [comment]
maaaaan calm down, you asked a question 11 hours ago, why are you closing the issue, give me the time to respond. is everything all right? You going through hard times?


Btw yes I have useSafeArea: false, this is my DropDown

```dart
DropdownSearch<String>(
                onChanged: (value) {
                  if (value != null) {
                    context.read<SchoolSearchBloc>().add(
                          SchoolSearchEvent.setCityFilter(value),
                        );
                  }
                },
                items: asyncSnapshot.data!,
                compareFn: (i, s) => i == s,
                dropdownButtonProps: const DropdownButtonProps(
                  padding: EdgeInsets.zero,
                ),
                dropdownDecoratorProps: DropDownDecoratorProps(
                  baseStyle: AppTextStyles.splashPhoneNumberInput.copyWith(
                    fontSize: 14,
                  ),
                  textAlign: TextAlign.left,
                  textAlignVertical: TextAlignVertical.center,
                  dropdownSearchDecoration: InputDecoration(
                    hintText: 'Comune',
                    hintStyle: AppTextStyles.splashPhoneNumberInput.copyWith(
                      color: Colors.black26,
                      fontSize: 14,
                    ),
                    contentPadding: EdgeInsets.zero,
                    border: InputBorder.none,
                  ),
                ),
                popupProps: PopupPropsMultiSelection.modalBottomSheet(
                  isFilterOnline: true,
                  modalBottomSheetProps: ModalBottomSheetProps(


                    useSafeArea: false, //                <--------------       HERE


                  ),
                  showSelectedItems: true,
                  showSearchBox: true,
                  searchFieldProps: TextFieldProps(
                    textInputAction: TextInputAction.go,
                    autofocus: true,
                    style: AppTextStyles.splashPhoneNumberInput,
                    decoration: InputDecoration(
                      hintText: 'Cerca comune',
                      hintStyle: AppTextStyles.splashPhoneNumberInput.copyWith(
                        color: Colors.black26,
                      ),
                      contentPadding: EdgeInsets.zero,
                    ),
                  ),
                  itemBuilder: (
                    BuildContext context,
                    String? cityName,
                    bool isSelected,
                  ) =>
                      CityNameTile(
                    isSelected: isSelected,
                    cityName: cityName ?? '',
                  ),
                ),
              )
```

**Comment by unsalu** [comment]
Hi Salim,
when i use modalBottomSheet it is really not working when keaybord is opened. 
I also test your example, it is not working.

I also try useSafeArea=false options. But it did not solved the problem. 

Please help us how we can solve this problem.

Best Regards.

![picture](https://user-images.githubusercontent.com/15001939/225011875-a982a9a9-17fb-4519-a3ab-836cb3ded2b4.JPG)

**Comment by mariopepe** [comment]
@salim-lachdhaf hey man everything all right?

**Comment by DumbDev168** [comment]
A work around solution here. You should be able to solve it by adding padding `EdgeInsets.only(bottom: MediaQuery.of(context).viewInsets.bottom)` to `showModalBottomSheet` function, more info https://github.com/salim-lachdhaf/searchable_dropdown/blob/fefe0735d56fcc11189807ca8e77497073a3ffea/lib/dropdown_search.dart#L632

In case this widget is a nested child, MediaQuery could be unable to re-render the viewinset value. You could use this custom scaffold instead of normal scaffold :

```
import 'package:flutter/material.dart';

/// A Scaffold to avoid viewInsets bottom not updated on nested child
/// Solve keyboard on has padding when modalBottomSheet textinput is focused
class AppScaffoldViewInset extends StatelessWidget {
  const AppScaffoldViewInset({
    required this.child,
    super.key,
  });

  final Widget child;

  @override
  Widget build(BuildContext context) {
    final viewInsetBottom = MediaQuery.of(context).viewInsets.bottom;
    return Scaffold(
      resizeToAvoidBottomInset: false,
      body: Container(
        height: double.infinity,
        alignment: Alignment.bottomCenter,
        padding: EdgeInsets.only(
          bottom: viewInsetBottom > 0 ? viewInsetBottom : 0,
        ),
        child: child,
      ),
    );
  }
}

```

**Comment by Miguel-A-Jara** [comment]
Adding to @DumbDev168, I found that this issue was introduced in f66ce1c. 

More specifically, by removing the Container that was wrapping `_popupWidgetInstance()`.

## FROM (5.0.4)

```dart
///open BottomSheet (Dialog mode)
  Future _openModalBottomSheet() {
    final sheetTheme = Theme.of(context).bottomSheetTheme;
    return showModalBottomSheet<T>(
    // ...
    builder: (ctx) {
      return Container(
        margin: widget.popupProps.modalBottomSheetProps.padding,
        child: _popupWidgetInstance(),
      );
    });
  }
```

## TO (5.0.5)
```dart
///open BottomSheet (Dialog mode)
  Future _openModalBottomSheet() {
    final sheetTheme = Theme.of(context).bottomSheetTheme;
    return showModalBottomSheet<T>(
    // ...
    builder: (ctx) =>  _popupWidgetInstance();
  )}
```


https://github.com/salim-lachdhaf/searchable_dropdown/commit/f66ce1c5dcd810deb21f55709e1280329997854f#diff-8905c8a5d0a1f4f37a23ac082cf564a7a34799a608ad86628f7eb3e6fe750d56L595

**Comment by andyngdz** [comment]
I'm also getting the same issue. Is there any fix around for now?

**Comment by salim-lachdhaf** [comment]
fixed in the next version v6.0.0-pre that will be released in next days
